vortex-array/src/stats/mod.rs

```diff
@@ -188,7 +188,9 @@ pub trait Statistics {
 pub trait ArrayStatistics {
     fn statistics(&self) -> &dyn Statistics;
 
-    fn inherit_statistics(&self, parent: &dyn Statistics);
+    fn take_stats(&self) -> Box<StatsSet>;
```

Comment:
the fact that this takes `&self` instead of `&mut self` is kind of sketch

Reply:
yeah, I understand how it works rn, but this is a `pub trait`... as API, it feels weird to imply that implementations should have interior mutability?

Reply:
yeah, I'm not sold on this api, just testing things right now. I have found some sketchy things in stats that I'm trying to make better but they require bigger changes